hw/ip/keymgr/doc/keymgr_testplan.hjson

```diff
@@ -0,0 +1,172 @@
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: op_at_wipe_state
+      desc: '''
+            `StWipe` is a temporary state during init operation, which will set CSR CFGEN.
+
+            Stimulus and checks:
+            Test command and reg access gated by CFGEN is ignored during `StWipe`.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: hw_sideload
+      desc: '''
+            Keymgr contains HW sideload interfaces to output keys for KMAC, HMAC, AES.
+
+            Stimulus:
+            Generate a keymgr output to HW sideload interface, excising all the sideload interfaces.
+
+            Checks:
+            Verify the sideload data and status for correctness.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: init_n_start
+      desc: '''
+            Stimulus and checks:
+            Verify all combinations of `init` and `start`.
+            - If both are set in `StReset`, check `INVALID_OP` error is triggered.
+            - If it's in other state, `init` is ignored
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: direct_to_disabled_state
+      desc: '''
+            Stimulus and checks:
+            Directly go to `StDisabled` from any state and check `StDisabled` is entered correctly.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: lc_disable
+      desc: '''
+            Life cycle can disable keymgr and let keymgr wipe secret immediately
+
+            Stimulus:
+            Test life cycle disables keymgr in any state.
+
+            Checks:
+            - If keymgr is not initialized, check it can't be initialized until life cycle enables
+              keymgr.
+            - If keymgr is in a valid state after init, key output to KMAC is wiped immediately and
+              SW output will be invalid after OP is done
+            - If keymgr in disabled state, check the behavior is consistent with normal behavior.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_cmd
+      desc: '''
+            Key manager has some security checks for fault injection, and it triggers an
+            `INVALID_CMD` error. This isn't possible to be raised by SW.
+
+            Stimulus:
+            Force one of these or both
+            - force advance, gen_id and gen_out to not one-hot
+            - force KMAC FSM to default branch to trigger FSM error
+
+            Checks:
+            - Check interrupts `err` is triggered
+            - Check alert `fault_err` is triggered and err_code is `INVALID_CMD`
+            - Check KMAC output key and output data to SW are corrupted, also keymgr enters
+              StDisabled
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: kmac_error
+      desc: '''
+            KMAC can respond error to trigger an `INVALID_CMD` error when keymgr sends data to it
```

Comment:
```suggestion
            Verify `keymgr` behavior on error response received from `KMAC` when attempting to send data to it.  
```

The goal indicates what is tested very briefly at a high level. What exactly is expected to happen is covered in 'checks' part.

Reply:
Updated.